### ISSUE TITLE
Dates for checkins & checkouts only

### ISSUE THREAD

**Issue by rvfx** [wakirin/Litepicker#173]
**Is your feature request related to a problem? Please describe.**
I want to use the Litepicker for a holiday home where there are three options for a specific date:
Checkin & Checkout, Checkin only, Checkout only.

I tried to implement this functionality with the lockDaysFilter option, but it's not the best user experience to get a warning _after_ trying to select a date.

**Describe the solution you'd like**
The best solution (for my situation) would be three arrays (lockDays, noCheckinDays, noCheckoutDays), but I know that's too specific.

Another good solution would be a live update of the `is-locked` class to directly show locked days when a specific `date1` is selected.

**Describe alternatives you've considered**
I can use the `tooltip` event to "warn" the user.

**Additional context**
If supporting the project would speed up the implementation of a solution, I would be happy to help.

**Comment by wakirin** [comment]
> I tried to implement this functionality with the lockDaysFilter option, but it's not the best user experience to get a warning after trying to select a date.

Could you show what you have tried?

**Comment by rvfx** [comment]
I only made a quick test with this code:
```
lockDaysFilter: function(checkin, checkout, pickedDates){
    if (checkout && checkout.getDate() == 10) {
        true
    } else {
        false
    }
```
It correctly disallows selecting `date2` if that's the 10th, but I get no visual feedback after selecting `date1`. I just get the `error:range` feedback _after_ I tried to select a wrong `date2`.

I found this calendar which initially marks checkin days and marks possible checkout days when `date1` is selected. For example: The 16th can't be selected for `date1` here:

![Bildschirmfoto 2021-02-02 um 09 35 20](https://user-images.githubusercontent.com/6519355/106574290-05388c80-653b-11eb-84c9-830e0c12beed.png)

After I click on the 8th, the 16th can be selected for `date2`. (There is also some minStay going on.)

![Bildschirmfoto 2021-02-02 um 09 43 35](https://user-images.githubusercontent.com/6519355/106574403-28633c00-653b-11eb-92c5-683a4a9f4fa3.png)

**Comment by wakirin** [comment]
> It correctly disallows selecting date2 if that's the 10th, but I get no visual feedback after selecting date1. I just get the error:range feedback after I tried to select a wrong date2.

I don't understand your example very well, what is the check-in date in your example? Why disallow 10th ?

> After I click on the 8th, the 16th can be selected for date2. (There is also some minStay going on.)

So you want to pick only 7 days? 
Or could it be in the range of 1 to 7 days and no more?

**Comment by mijrs** [comment]
> After I click on the 8th, the 16th can be selected for `date2`. (There is also some minStay going on.)

Can you provide a demo on jsfiddle.net?

And great design! Can you also share this in the jsfiddle.net?

**Comment by strarsis** [comment]
I had to implement the same feature: The allowed check-in dates are different to the check-out dates: Only few dates may be allowed for check-in (like only every monday on summer season).
I implemented this with Litepicker by checking whether no date or the start date was selected. When no date was selected, the check-in date restrictions are enforced, when a start date has been selected, the check-out date restrictions were applied instead.
Related: https://github.com/wakirin/Litepicker/issues/116
